src/cryptoManager.ts

```diff
@@ -36,10 +39,9 @@ export class CryptoManager {
     }
   }
 
-  private async computeHash(filePath: string): Promise<Buffer> {
-    const secret = await fsp.readFile(filePath, { encoding: 'binary' });
+  private computeHash(bufferAsString: string): Buffer {
     const hash = crypto.createHash(this.cryptoConfig.shaSize);
-    hash.update(String(secret));
+    hash.update(bufferAsString);
```

Comment:
hash update accepts buffer as well as string using the buffer directly is better then converting it to string as it converts it back to buffer internally